src/main/webapp/resources/js/cvoc-interface.js

```diff
@@ -0,0 +1,55 @@
+// depends on jQuery ajax
+var mapquery = 'prefLabel';
+var mapid = 'uri';
+var mapping = { query: mapquery,  id: mapid };
+
+// autocomplete calls this
+function autointerface(request, response, cv, mapping) {
+    var protocol = cv.protocol;
+    if (!protocol) { protocol = 'skosmos'; };//default
+    if (protocol == 'skosmos') {
+        return(skosmos(request, response, cv, mapping)); }
+    if (protocol == 'example') {
+        return(autoexample(request, response)); };
+};
+
+function skosmos(request, response, cv, mapping) {
+    var termParentUri = "";
+    if (cv.termParentUri != "")
+        termParentUri = "&parent=" + cv.termParentUri;
+    var result = [];
+    var tmp = $.ajax({
```

Comment:
@poikilotherm, the interface could be implemented using any popular Javascript framework, it should be placed preferably outside of the Dataverse core.

Reply:
@4tikhonov maybe I didn't make my point clear enough: the Fetch API is a browser builtin Javascript API around since ~2015. These days, it's supported everywhere and makes using a framework obsolete. 

By using the W3C standardised web API implemented in any browser, you can completely avoid relying on a framework, which makes your JS scripts extremly portable and compatible with any framework you're using. There is no need to reimplement or change the code to reuse it when using different frameworks.

This is in no way using sth. like a Dataverse REST endpoint or anything like it - all of this happens client side in your browser. I hope this gets a bit clearer now. :smile_cat:

Reply:
The cvoc-interface.js is just client side code included with a script element. When placed outside of the Dataverse code it is more like configuration of the connection to the external services (only SKOSMOS for now). We complied it in the war file just to have a default implementation that works without introducing the extra work of serving a json resource on the server, with the added apache conf settings changes. When externalising we would have to do a bit more explaining of the 'interface', which now has to be inferred from the code.

Reply:
> The cvoc-interface.js is just client side code included with a script element. 

Aye.

> When placed outside of the Dataverse code it is more like configuration of the connection to the external services (only SKOSMOS for now).

Wait - "like configuration"? The configuration this code uses to interface with the external service is stored inside Dataverse (the database). I'm not sure what you mean here then? As far as I understood it, you are trying to follow the code against interface principle, which is great.

> We complied it in the war file just to have a default implementation that works without introducing the extra work of serving a json resource on the server, with the added apache conf settings changes.

Yes please, it's good to have a default in place which serves the most usual use cases. Thank you for this!

> When externalising we would have to do a bit more explaining of the 'interface', which now has to be inferred from the code.

That would be marvelous! Maybe you could add sth to the developer guide of Dataverse?

From your comment I still don't understand why relying on the jQuery UI framework is a necessity for this AJAX call, when the same functionality is provided by any (recent) browser out there, allowing better reuse and less bundled dependencies.

Reply:
Hi @poikilotherm, just for your information, we've originally placed the interface outside of Dataverse core to make it more flexible https://github.com/Dans-labs/semantic-gateway/blob/main/static/js/interface.js